### PR TITLE
Upgrade to fastify 3

### PR DESCRIPTION
**Pull request by kibertoad**
#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,10 @@ For instance:
 ```js
 const { fastifyRequestContextPlugin } = require('fastify-request-context')
 const fastify = require('fastify');
+const middie = require('middie')
 
 const app = fastify({ logger: true })
+app.register(middie)
```
Comment (mcollina) [inline README.md:52]: Is this needed?

File: test/requestContextPlugin.spec.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
   const app = fastify({ logger: true })
-  app.register(fastifyRequestContextPlugin)
+  return app.register(middie).then(() => {
+    app.register(fastifyRequestContextPlugin)
 
-  app.get('/', endpoint)
+    app.get('/', endpoint)
 
-  return app.ready()
+    return app.ready().then(() => app)
```
Comment (mcollina) [inline test/requestContextPlugin.spec.js:13]: why this change? I think this is fixed `ready()` resolves to `app`.

Comment (kibertoad) [reply test/requestContextPlugin.spec.js:13]: Was the fix released? it was returning undefined for me before I did the change.

File: README.md (hunk 1 above)
Comment (kibertoad) [reply README.md:52]: Will try removing it, I thought middie is mandatory now for using plugins?..

File: test/requestContextPlugin.spec.js (hunk 2 above)
Comment (kibertoad) [reply test/requestContextPlugin.spec.js:13]: Updated to 3.0.3, that resolved it.

File: README.md (hunk 1 above)
Comment (kibertoad) [reply README.md:52]: Removed it, test still passes.

Comment (mcollina) [reply README.md:52]: middie is not mandatory. It's mandatory only if you want to use `.use()`, e.g. express-style middlewares.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fastify-request-context",
-  "version": "1.0.2",
+  "version": "2.0.0",
```
Comment (Eomm) [inline package.json:3]: Could you revert this change?

It is in conflicts with our release tool

File: README.md (hunk 1 above)
Comment (Eomm) [reply README.md:52]: is it adding chaos in the readme then?

File: package.json (hunk 3 above)
Comment (kibertoad) [reply package.json:3]: So we don't bump versions manually?

Comment (mcollina) [reply package.json:3]: we normally bump them directly on master on release. So multiple PRs can go in the same release.

**Comment by kibertoad** [comment]
@Eomm Thank you for the comments! Fixed.

**Comment by Eomm** [comment]
Right now I don't have power here.

Would you like to release it?

Using [releasify](https://github.com/fastify/releasify) it will easy as runs `releasify -s major -m -n --npm-otp 123456` 👍

**Comment by kibertoad** [comment]
@Eomm Thanks, will try to do so!
